### PR TITLE
User role getters refactor

### PR DESCRIPTION
**Pull request by Divinare**
Relates to: https://github.com/HSLdevcom/bultti/issues/96

### CODE REVIEW

File: src/inspection/InspectionActions.tsx [hunk 1]
```diff
@@ -174,7 +173,7 @@ const InspectionActions = observer(
     }, [onRefresh, inspection, disabledActions])
 
     let userCanPublish =
-      inspection.status === InspectionStatus.InReview && requireAdminUser(user)
+      inspection.status === InspectionStatus.InReview && useHasAdminAccessRights()
```
Comment (danieldunderfelt) [inline src/inspection/InspectionActions.tsx:176]: This is not valid React hooks usage as the hook will not be called if the first condition returns true. Instead, put the return value from the hook into a variable and use the variable here in the condition.

File: src/inspection/InspectionActions.tsx [hunk 2]
```diff
@@ -227,7 +226,7 @@ const InspectionActions = observer(
           )}
 
           {inspectionCanBeReady &&
-            requireOperatorUser(user, inspection?.operatorId || undefined) &&
+            useHasOperatorUserAccessRights(inspection?.operatorId || undefined) &&
```
Comment (danieldunderfelt) [inline src/inspection/InspectionActions.tsx:229]: Not valid React hook usage (see the previous comment).

File: src/inspection/InspectionActions.tsx [hunk 3]
```diff
@@ -241,7 +240,7 @@ const InspectionActions = observer(
           {[InspectionStatus.Draft, InspectionStatus.InProduction].includes(
             inspection.status
           ) &&
-            requireAdminUser(user) && (
+            useHasAdminAccessRights() && (
```
Comment (danieldunderfelt) [inline src/inspection/InspectionActions.tsx:243]: Not valid React hook usage (see the previous comment).

File: src/inspection/InspectionActions.tsx [hunk 4]
```diff
@@ -278,7 +277,7 @@ const InspectionActions = observer(
         </ButtonRow>
         {submitActive &&
           inspectionCanBeSubmitted &&
-          requireOperatorUser(user, inspection?.operatorId || undefined) &&
+          useHasOperatorUserAccessRights(inspection?.operatorId || undefined) &&
```
Comment (danieldunderfelt) [inline src/inspection/InspectionActions.tsx:280]: Not valid React hook usage (see the previous comment).

File: src/page/OperatorContractsListPage.tsx [hunk 5]
```diff
@@ -115,7 +114,7 @@ const OperatorContractsListPage = observer(({ children }: PropTypes) => {
           <MessageView>Ei sopimusehtoja.</MessageView>
         </MessageContainer>
       )}
-      {!!operator && requireAdminUser(user) && (
+      {!!operator && useHasAdminAccessRights() && (
```
Comment (danieldunderfelt) [inline src/page/OperatorContractsListPage.tsx:117]: Not valid React hook usage (see the previous comment).

**Review by danieldunderfelt — CHANGES_REQUESTED** [review]
I like the hooks, they are a good idea, but there are issues with how they are used in the code.

**Comment by Divinare** [comment]
Review fixes done. Now hooks are only called at the top level and not in conditions.

**Comment by Divinare** [comment]
Merging, the changes should be ok

**Pull request merged**